install.sh

```diff
@@ -1,25 +1,247 @@
+# if it needs to be changed.
+checkGnmicInstalledVersion() {
+    if [[ -f "${GNMIC_INSTALL_DIR}/${BINARY_NAME}" ]]; then
+        local version=$("${GNMIC_INSTALL_DIR}/${BINARY_NAME}" version | head -1 | awk '{print $NF}')
+        if [[ "v$version" == "$TAG" ]]; then
+            echo "gnmic is already ${DESIRED_VERSION:-latest}"
+            return 0
+        else
+            echo "gnmic ${TAG_WO_VER} is available. Changing from version ${version}."
+            return 1
+        fi
+    else
+        return 1
+    fi
+}
+
+# downloadFile downloads the latest binary package and also the checksum
+# for that binary.
+downloadFile() {
+    GNMIC_DIST="${BINARY_NAME}_${TAG_WO_VER}_${OS}_${ARCH}.tar.gz"
+    DOWNLOAD_URL="https://github.com/karimra/gnmic/releases/download/${TAG}/${GNMIC_DIST}"
+    CHECKSUM_URL="https://github.com/karimra/gnmic/releases/download/${TAG}/checksums.txt"
+    GNMIC_TMP_ROOT="$(mktemp -d)"
+    GNMIC_TMP_FILE="$GNMIC_TMP_ROOT/$GNMIC_DIST"
+    GNMIC_SUM_FILE="$GNMIC_TMP_ROOT/checksums.txt"
+    echo "Downloading $DOWNLOAD_URL"
+    if type "curl" &>/dev/null; then
+        curl -SsL "$CHECKSUM_URL" -o "$GNMIC_SUM_FILE"
+    elif type "wget" &>/dev/null; then
+        wget -q -O "$GNMIC_SUM_FILE" "$CHECKSUM_URL"
+    fi
+    if type "curl" &>/dev/null; then
+        curl -SsL "$DOWNLOAD_URL" -o "$GNMIC_TMP_FILE"
+    elif type "wget" &>/dev/null; then
+        wget -q -O "$GNMIC_TMP_FILE" "$DOWNLOAD_URL"
+    fi
+}
+
+# installFile verifies the SHA256 for the file, then unpacks and
+# installs it.
+installFile() {
+    GNMIC_TMP="$GNMIC_TMP_ROOT/$BINARY_NAME"
+    if [ $VERIFY_CHECKSUM == "true" ]; then
+        local sum=$(openssl sha1 -sha256 ${GNMIC_TMP_FILE} | awk '{print $2}')
+        local expected_sum=$(cat ${GNMIC_SUM_FILE} | grep -i $GNMIC_DIST | awk '{print $1}')
+        if [ "$sum" != "$expected_sum" ]; then
+            echo "SHA sum of ${GNMIC_TMP_FILE} does not match. Aborting."
+            exit 1
+        fi
+    fi
+
+    mkdir -p "$GNMIC_TMP"
+    tar xf "$GNMIC_TMP_FILE" -C "$GNMIC_TMP"
+    GNMIC_TMP_BIN="$GNMIC_TMP/gnmic"
+    echo "Preparing to install $BINARY_NAME ${TAG_WO_VER} into ${GNMIC_INSTALL_DIR}"
+    runAsRoot cp "$GNMIC_TMP_BIN" "$GNMIC_INSTALL_DIR/$BINARY_NAME"
+    echo "$BINARY_NAME ${TAG_WO_VER} installed into $GNMIC_INSTALL_DIR/$BINARY_NAME"
+}
+
+# fail_trap is executed if an error occurs.
+fail_trap() {
+    result=$?
+    if [ "$result" != "0" ]; then
+        if [[ -n "$INPUT_ARGUMENTS" ]]; then
+            echo "Failed to install $BINARY_NAME with the arguments provided: $INPUT_ARGUMENTS"
+            help
+        else
+            echo "Failed to install $BINARY_NAME"
+        fi
+        echo -e "\tFor support, go to https://github.com/karimra/gnmic"
+    fi
+    cleanup
+    exit $result
+}
+
+# testVersion tests the installed client to make sure it is working.
+testVersion() {
+    set +e
+    HELM="$(command -v $BINARY_NAME)"
```

Comment:
This part is failing for me.
```bash
[root@playground ~]# sudo curl -sL https://raw.githubusercontent.com/karimra/gnmic/new-installer/install.sh | sudo bash 
gnmic 0.1.1 is available. Changing from version 0.1.0.
Downloading https://github.com/karimra/gnmic/releases/download/v0.1.1/gnmic_0.1.1_linux_x86_64.tar.gz
Preparing to install gnmic 0.1.1 into /usr/local/bin
gnmic 0.1.1 installed into /usr/local/bin/gnmic
gnmic not found. Is /usr/local/bin on your $PATH?
Failed to install gnmic
	For support, go to https://github.com/karimra/gnmic
```

Reply:
I can't reproduce this, though I happened to this this error on this very line as well.
What I will do is that I will substitute this command with `gnmic version` to test that the installed program is working

Reply:
resolved in 01d9af6